src/main/java/sg/ncl/MainController.java

```diff
@@ -725,53 +723,53 @@ public String signup2(Model model, HttpServletRequest request) {
+            @ModelAttribute(SIGNUP_MERGED_FORM) SignUpMergedForm signUpMergedForm,
             BindingResult bindingResult,
             final RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
 
-        if (bindingResult.hasErrors() || signUpMergedForm.getIsValid() == false) {
+        if (bindingResult.hasErrors() || !signUpMergedForm.getIsValid()) {
             log.warn("Register form has errors {}", signUpMergedForm.toString());
-            return "signup2";
+            return SIGNUP_PAGE;
         }
 
         if (!signUpMergedForm.getHasAcceptTeamOwnerPolicy()) {
             signUpMergedForm.setErrorTeamOwnerPolicy("Please accept the team owner policy");
             log.warn("Policy not accepted");
-            return "signup2";
+            return SIGNUP_PAGE;
         }
 
         // get form fields
         // craft the registration json
         JSONObject mainObject = new JSONObject();
         JSONObject credentialsFields = new JSONObject();
         credentialsFields.put("username", signUpMergedForm.getEmail().trim());
-        credentialsFields.put("password", signUpMergedForm.getPassword());
+        credentialsFields.put(PSWD, signUpMergedForm.getPassword());
 
         // create the user JSON
         JSONObject userFields = new JSONObject();
         JSONObject userDetails = new JSONObject();
         JSONObject addressDetails = new JSONObject();
 
-        userDetails.put("firstName", signUpMergedForm.getFirstName().trim());
-        userDetails.put("lastName", signUpMergedForm.getLastName().trim());
-        userDetails.put("jobTitle", signUpMergedForm.getJobTitle().trim());
-        userDetails.put("email", signUpMergedForm.getEmail().trim());
-        userDetails.put("phone", signUpMergedForm.getPhone().trim());
-        userDetails.put("institution", signUpMergedForm.getInstitution().trim());
-        userDetails.put("institutionAbbreviation", signUpMergedForm.getInstitutionAbbreviation().trim());
-        userDetails.put("institutionWeb", signUpMergedForm.getWebsite().trim());
-        userDetails.put("address", addressDetails);
+        userDetails.put(FNAME, signUpMergedForm.getFirstName().trim());
+        userDetails.put(LNAME, signUpMergedForm.getLastName().trim());
+        userDetails.put(JOB_TITLE, signUpMergedForm.getJobTitle().trim());
+        userDetails.put(EMAIL, signUpMergedForm.getEmail().trim());
+        userDetails.put(PHONE, signUpMergedForm.getPhone().trim());
+        userDetails.put(INSTITUTION, signUpMergedForm.getInstitution().trim());
+        userDetails.put(INSTITUTION_ABBREVIATION, signUpMergedForm.getInstitutionAbbreviation().trim());
+        userDetails.put(INSTITUTION_WEB, signUpMergedForm.getWebsite().trim());
+        userDetails.put(ADDRESS, addressDetails);
 
         addressDetails.put("address1", signUpMergedForm.getAddress1().trim());
         addressDetails.put("address2", signUpMergedForm.getAddress2().trim());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "address2" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)